pkg/controllers/gameserverset/gameserverset_manager.go

```diff
@@ -364,6 +366,7 @@ func (manager *GameServerSetManager) SyncStatus() error {
 		MaintainingReplicas:     pointer.Int32Ptr(int32(maintainingGs)),
 		WaitToBeDeletedReplicas: pointer.Int32Ptr(int32(waitToBeDeletedGs)),
 		LabelSelector:           asts.Status.LabelSelector,
+		ObservedGeneration:      gss.GetGeneration(),
 	}
 
```

Comment:
Please whether the old and new status is equal, If no changed, do not need to patch.